### DISCUSSION TITLE
How to store `ws` connection to Redis

### DISCUSSION THREAD

**Discussion by StarMonk** [uNetworking/uWebSockets.js#846] [Q&A] [answered]
I am trying to store the `ws` connection of each user in REDIS, but whenever I stringify the `ws` I get an empty value stored in there.

Do we have the option to store `ws` connection in REDIS and Fetch and use them to send messages?

```
client.sAdd("user_id", JSON.stringify(ws), (err) => {
            if (err) {
                console.error(err);
            }
        });
```

I can't store `ws` directly as redis throw an encoder error, I am looking for an alternative to store them to REDIS for vertical scaling.

**Comment by e3dio** [comment, marked as answer]
Store the ID

**Reply by StarMonk** [reply]
Do you mean `ws.id` that is created using `upgrade`  [here](https://github.com/uNetworking/uWebSockets.js/discussions/843#discussioncomment-4463790) ? 

Even if I store this ID, I still have to do `wsMap` and store it in `Map` on the server, so this doesn't make sense. Cause my ID comes from MySQL database and I already have one. But I don't have associated `webSocket client` for the id and when I was trying to store websocket client reference in REDIS, redis complained about the data type, I was storing, so I did JSON.stringify and found out it doesn't store in uWS.websocket format

How I am gonna send a request with that `ID`?  Can you clarify, please?

Thanks

**Reply by e3dio** [reply]
> How I am gonna send a request with that ID? Can you clarify, please?

```javascript
wsMap.get(id).send('message');
```

**Reply by StarMonk** [reply]
yup, so that would mean, I will have to maintain `wsMap` locally, and will have to scale horizontally mostly.
